packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNode.ts

```diff
@@ -165,26 +180,24 @@ export class SummarizerNode implements IRootSummarizerNode {
 		trackState: boolean = true,
 		telemetryContext?: ITelemetryContext,
 	): Promise<ISummarizeResult> {
+		// Set to wipSummarizeCalled true to represent that current node was included in the summary process.
+		this.wipSummarizeCalled = true;
+
 		// If trackState is false, call summarize internal directly and don't track any state.
 		if (!trackState) {
 			return this.summarizeInternalFn(fullTree, trackState, telemetryContext);
 		}
 
 		// Try to reuse the tree if unchanged
 		if (this.canReuseHandle && !fullTree && !this.hasChanged()) {
-			const latestSummary = this._latestSummary;
-			if (latestSummary !== undefined) {
-				this.wipLocalPaths = {
-					localPath: latestSummary.localPath,
-					additionalPath: latestSummary.additionalPath,
-				};
+			if (this._lastSummaryReferenceSequenceNumber !== undefined) {
 				this.wipSkipRecursion = true;
 				const stats = mergeStats();
 				stats.handleNodeCount++;
 				return {
 					summary: {
 						type: SummaryType.Handle,
-						handle: latestSummary.fullPath.path,
+						handle: this._summaryHandleId.toString(),
```

Comment:
The ids of data stores and DDSes can now be special characters which will get encoded. Please make sure that the encoding happens correctly such that it matches with the ids in the summary tree.
For instance, say a DDS id is `/@/&`. The summary tree will have a sub-tree for data store with id `@` (or the encoded format) and a sub-tree for a DDS under it with id `&` (or the encoded format). The handle is here should match the id for that tree.
It looks like you kept the logic the same, but it's hard to tell from the code and we had a bug like that already in the driver code.

Reply:
Added a new test file - `summarizerNodeUtils.spec.ts`